dotcom-rendering/src/server/index.article.web.ts

```diff
@@ -30,13 +33,23 @@ const enhanceBadge = (badge?: FEArticleBadgeType) =>
 		: undefined;
 
 export const enhanceArticleType = (body: unknown): FEArticleType => {
-	const data = validateAsArticleType(body);
+	const validated = validateAsArticleType(body);
+	// addImageIDs needs to take account of both main media elements
+	// and block elements, so it needs to be executed here
+	const { mainMediaElements, blocks } = addImageIDs(validated);
+	const data = {
+		...validated,
+		mainMediaElements,
+		blocks,
+	};
```

Comment:
Why create `data` here? I would have thought `blocks` and `mainMediaElements` would be passed to `enhanceBlocks` and `addLightBoxData` respectively, and avoid creating an unused object.

Reply:
Please address this 🙏